package.json

```diff
@@ -2,11 +2,12 @@
   "name": "envalid",
   "version": "4.1.4",
   "description": "Validation for your environment variables",
-  "main": "index.js",
+  "main": "src/envalid.js",
+  "types": "src/envalid.d.ts",
+  "browser": "src/envalidWithoutDotenv.js",
+  "react-native": "src/envalidWithoutDotenv.js",
```

Comment:
could you point to some docs showing that this is a documented feature? It seems awesome, but we've never gotten a request for supporting it in Jest (Jest supports `browser`), so I haven't seen it before 🙂

Reply:
I saw the comment about `_require` and thought it would be better if React Native could use the same file as `browser`. Somehow I ended up finding [stereobooster/package.json](https://github.com/stereobooster/package.json), which [documents it](https://github.com/stereobooster/package.json#react-native). As someone who has used React Native a fair bit I also wasn't aware of it before this.

The code for metro on master makes it a bit less clear than the hash that page links to. Going off what it links to it looks like RN might choose `browser` if `react-native` isn't there, in which case just adding `browser` would work. I think being explicit is better though.

<img width="1920" alt="envalid" src="https://user-images.githubusercontent.com/1537615/51821093-f52fa780-22a5-11e9-8e38-527ab4d408c7.png">

It actually looks like React Native support won't currently work because this package uses `url`. I've replaced it with `universal-url`, which seems to be a decent package for it.

Reply:
>I guess because it doesn't seem to be officially documented anywhere.

Might be worth opening up an issue with metro. Or have a file called `index.native.js` or something

Reply:
Thinking about it slightly more. It makes sense that React Native would use `browser` since it mostly emulates a browser environment. I'd be fine removing the `react-native` field and just leaving it at that.

I think it should actually be documented on React Native instead of in Metro, since React Native should now probably be using [`resolverMainFields`](https://facebook.github.io/metro/docs/en/configuration#resolvermainfields), instead of relying on [hard coding](https://github.com/facebook/metro/blob/master/packages/metro/src/ModuleGraph/node-haste/Package.js#L108).

Also I'm open to alternative `url` replacements, assuming we want to actually have RN working after this PR something would have to be done though.

Reply:
I like `universal-url`, seems perfect.